contrib/drivers/huawei/dorado/metrics.go

```diff
@@ -0,0 +1,232 @@
+	metricToUnitMap := make(map[string]string)
+	for _, resources := range configs.Cfgs {
+		switch resources.Resource {
+		//ToDo: Other Cases needs to be added
+		case "volume", "pool", "controller":
+			for index, metricName := range resources.Metrics {
+
+				metricToUnitMap[metricName] = resources.Units[index]
+
+			}
+		}
+	}
+	return metricToUnitMap
+}
+
+// 	getMetricList:- is  to get the list of supported metrics for given resource type
+//	supportedMetrics -> list of supported metrics
+func (d *MetricDriver) GetMetricList(resourceType string) (supportedMetrics []string, err error) {
+	var configs Configs
+
+	//Read supported metric list from yaml config
+	source := []byte(data)
+	err = yaml.Unmarshal(source, &configs)
+	if err != nil {
+		log.Fatalf("unmarshal error: %v", err)
+	}
+
+	for _, resources := range configs.Cfgs {
+		if resources.Resource == resourceType {
+			switch resourceType {
+			case "volume", "pool", "controller":
+				for _, m := range resources.Metrics {
+					supportedMetrics = append(supportedMetrics, m)
+
+				}
+			}
+		}
+	}
+
+	return supportedMetrics, nil
+}
+
+//	CollectMetrics: Driver entry point to collect metrics. This will be invoked by the dock
+//	[]*model.MetricSpec	-> the array of metrics to be returned
+func (d *MetricDriver) CollectMetrics() ([]*model.MetricSpec, error) {
+
+	// get Metrics to unit map
+	metricToUnitMap := getMetricToUnitMap()
+	//validate metric support list
+	supportedMetrics, err := d.GetMetricList("volume")
+	if supportedMetrics == nil {
+		log.Infof("no metrics found in the  supported metric list")
+	}
+	controllers, err := d.client.ListControllers()
+	if err != nil {
+		log.Errorf("get controller failed: %s", err)
+		return nil, err
+	}
+
+	var tempMetricArray []*model.MetricSpec
+	for _, controller := range controllers {
+		// TODO: the controller id need to be optional
+		uuid := ObjectTypeController + ":" + controller.Id
+		dataIdList := []string{PerfIOPS, PerfReadThroughput, PerfWriteThroughput,
+			PerfResponseTime, PerfServiceTime, PerfUtilizationPrcnt}
+		metricMap, err := d.client.GetPerformance(uuid, dataIdList)
+		if err != nil {
+			log.Errorf("get performance data failed: %s", err)
+			return nil, err
+		}
+
+		name2id := map[string]string{
+			"iops":              PerfIOPS,
+			"read_throughput":   PerfReadThroughput,
+			"write_throughput":  PerfWriteThroughput,
+			"response_time":     PerfResponseTime,
+			"service_time":      PerfServiceTime,
+			"utilization_prcnt": PerfUtilizationPrcnt,
+		}
+		for _, element := range supportedMetrics {
+			val, _ := strconv.ParseFloat(metricMap[name2id[element]], 64)
+			metricValue := &model.Metric{
+				Timestamp: getCurrentUnixTimestamp(),
+				Value:     val,
+			}
+			metricValues := make([]*model.Metric, 0)
+			metricValues = append(metricValues, metricValue)
+			metric := &model.MetricSpec{
+				InstanceID:   uuid,
+				InstanceName: uuid,
+				Job:          "HuaweiOceanStor",
+				Labels:       map[string]string{},
```

Comment:
Please add some important properties related to the metrics in to label map. at least add  component name.